### PR TITLE
Added liveness and readiness probe for leader election sidecar container

### PR DESCRIPTION
**Pull request by lucferbux**
---
Title: Added liveness and readiness probe for leader election sidecar container
---

**Changes proposed in this pull request**
* Added liveness probe to leader election sidecar container
* Added readiness probe for leader election sidecar container

### CODE REVIEW

File: jupyterhub/jupyterhub/base/jupyterhub-dc.yaml [hunk 1]
```diff
@@ -209,9 +228,13 @@ spec:
...
-          defaultMode: 420
-          name: jupyterhub-cfg
-        name: config
+      - name: config 
+        configMap:
+            defaultMode: 420
+            name: jupyterhub-cfg 
+      - name: sidecar-script
+        configMap:
+          defaultMode: 511
```
Comment (maulikjs) [inline jupyterhub/jupyterhub/base/jupyterhub-dc.yaml:237]: Just one tiny comment, any specific reason this is mode 511?

**Review by anishasthana — APPROVED** [review]
/lgtm

File: jupyterhub/jupyterhub/base/jupyterhub-dc.yaml (hunk 1 above)
Comment (lucferbux) [reply jupyterhub/jupyterhub/base/jupyterhub-dc.yaml:237]: Yeah, this one is because by default there is no execution permission for the script.

**Pull request merged**